compiler/circle-interpreter-cffi-test/infer.py

```diff
@@ -0,0 +1,116 @@
+import argparse
+import h5py
+import numpy as np
+from pathlib import Path
+import re
+import sys
+
+############ Managing paths for the artifacts required by the test.
+
+
+def extract_test_args(s):
+    p = re.compile('eval\\((.*)\\)')
+    result = p.search(s)
+    return result.group(1)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--lib_path', type=str, required=True)
+parser.add_argument('--test_list', type=str, required=True)
+parser.add_argument('--artifact_dir', type=str, required=True)
+args = parser.parse_args()
+
+with open(args.test_list) as f:
+    contents = [line.rstrip() for line in f]
+comment_removed = [line for line in contents if not line.startswith('#')]
+newline_removed = [line for line in comment_removed if line.startswith('eval(')]
```

Comment:
comment_removed seems redundant.

```suggestion
contents = [line for line in contents if line.startswith('eval(')]
```

Reply:
I added this in advance to make it possible to have comment in `test.lst`.

Reply:
comment would be automatically removed with `if line.startswith('eval(')`.

Reply:
Oh, I mean in-line comment like this.
```
eval(~~~) # note this test is ~
```